src/core/hle/kernel/k_memory_region_type.h

```diff
@@ -0,0 +1,339 @@
+// Copyright 2021 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "common/bit_util.h"
+#include "common/common_funcs.h"
+#include "common/common_types.h"
+
+#define ARCH_ARM64
+#define BOARD_NINTENDO_NX
+
+namespace Kernel {
+
+enum KMemoryRegionType : u32 {
+    KMemoryRegionAttr_CarveoutProtected = 0x04000000,
+    KMemoryRegionAttr_DidKernelMap = 0x08000000,
+    KMemoryRegionAttr_ShouldKernelMap = 0x10000000,
+    KMemoryRegionAttr_UserReadOnly = 0x20000000,
+    KMemoryRegionAttr_NoUserMap = 0x40000000,
+    KMemoryRegionAttr_LinearMapped = 0x80000000,
+};
+DECLARE_ENUM_FLAG_OPERATORS(KMemoryRegionType);
+
+namespace impl {
+
+constexpr size_t BitsForDeriveSparse(size_t n) {
+    return n + 1;
+}
+
+constexpr size_t BitsForDeriveDense(size_t n) {
+    size_t low = 0, high = 1;
+    for (size_t i = 0; i < n - 1; ++i) {
+        if ((++low) == high) {
+            ++high;
+            low = 0;
+        }
+    }
+    return high + 1;
+}
+
+class KMemoryRegionTypeValue {
+private:
+    using ValueType = typename std::underlying_type<KMemoryRegionType>::type;
```

Comment:
```suggestion
    using ValueType = std::underlying_type_t<KMemoryRegionType>;
```

This should likely be public as well, given the public accessor that retrieves the value of this class uses it.